File: WebListIntent/app/src/main/java/com/dstrube/weblistintent/view/LinkAdapter.java
package com.dstrube.weblistintent.view;

import java.util.List;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.dstrube.weblistintent.R;
import com.dstrube.weblistintent.model.LinkData;

public class LinkAdapter extends ArrayAdapter<LinkData> {
	private Context ctx;
	private List<LinkData> linkDataList;

	public LinkAdapter(List<LinkData> linkDataList, Context ctx) {
		super(ctx, R.layout.row_layout, linkDataList);
		this.ctx = ctx;
		this.linkDataList = linkDataList;
	}

	@Override
	public int getCount() {
		return linkDataList.size();
	}

	@Override
	public LinkData getItem(int position) {
		return linkDataList.get(position);
	}

	@Override
	public long getItemId(int position) {
		return linkDataList.get(position).hashCode();
	}
	
	@Override
    public View getView(int position, View convertView, ViewGroup parent) {
       View v = convertView;
       if (convertView == null) {
                               LayoutInflater inflater = (LayoutInflater) ctx.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
                               v = inflater.inflate(R.layout.row_layout, null);
       }
       
       TextView linkDomainText = (TextView) v.findViewById(R.id.domain);
//       TextView lName = (TextView) v.findViewById(R.id.url);
       
       LinkData ld = linkDataList.get(position);
       linkDomainText.setText(ld.getDomain());
//       lName.setText(ld.getUrl());
       
       return v;
    
    }
}

File: kotlin/fix/WebListIntent/app/src/main/java/com/dstrube/weblistintent/view/LinkAdapter.kt
package com.dstrube.weblistintent.view

import android.content.Context
import com.dstrube.weblistintent.model.LinkData
import android.widget.TextView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.dstrube.weblistintent.R


class LinkAdapter(private val linkDataList: List<LinkData>, private val ctx: Context) : ArrayAdapter<LinkData>(ctx, R.layout.row_layout, linkDataList) {

    override fun getCount(): Int {
        return linkDataList.size
    }

    override fun getItem(position: Int): LinkData? {
        return linkDataList[position]
    }

    override fun getItemId(position: Int): Long {
        return linkDataList[position].hashCode().toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var v = convertView
        if (convertView == null) {
            val inflater = ctx.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            v = inflater.inflate(R.layout.row_layout, null)
        }

        val linkDomainText = v!!.findViewById(R.id.domain) as TextView
        //       TextView lName = (TextView) v.findViewById(R.id.url);

        val ld = linkDataList[position]
        linkDomainText.text = ld.domain
        //       lName.setText(ld.getUrl());

        return v

    }
}